kinematic_chain_base_link: yumi_base_link
contact_point_offset: 0.085
# contact_point_offset: 0.21
action_name: folding_controller
use_computed_torque: false
use_ground_truth_pc: false
loop_rate: 250

kdl_manager:
  eps: 0.005
  max_tf_attempts: 10
  ikvel_solver: nso
  nso_weight: 40

rod_arm:
  kdl_eef_frame: "yumi_link_7_l"
  gripping_frame: "l_gripping_point" #"l_gripping_point" #"gripper_l_finger_l"
  has_ft_sensor: true
  sensor_frame: "optodaq_l_sensor_link"
  sensor_topic: "/yumi/optodaq_l_compensated"
  sensor_calib:
    data: [1.0, 0, 0, 0, 0, 0,
           0, 1.0, 0, 0, 0, 0,
           0, 0, 1.0, 0, 0, 0,
           0.03321, 0.03192, 0.01945, 1.635, 0.3909, 0.4423,
           0.07226, -0.02405, -0.008477, 0.02264, 2.037, 1.466,
           -0.001555, 0.01690, 0.002053, 0.1, -0.1903, 1.324]
    # data: [1.0, 0, 0, 0, 0, 0,
    #        0, 1.0, 0, 0, 0, 0,
    #        0, 0, 1.0, 0, 0, 0,
    #        0.05975, 0.2789, -0.008671, 0.5075, 0.4840, -0.5221,
    #        -0.07749, -0.2948, -0.05920, 1.859, 1.966, -0.7643,
    #        0.04118, -0.001649, -0.002709, 0.2350, 0.2619, -0.04483]

surface_arm:
  kdl_eef_frame: "yumi_link_7_r"
  gripping_frame: "r_gripping_point"
  has_ft_sensor: true
  sensor_frame: "optodaq_r_sensor_link"
  sensor_topic: "/yumi/optodaq_r_compensated"
  sensor_calib:
    data: [1.0, 0, 0, 0, 0, 0,
           0, 1.0, 0, 0, 0, 0,
           0, 0, 1.0, 0, 0, 0,
           -0.04439, 0.00957, 0.02209, 1.909, -0.2187, 1.185,
           0.03042, -0.0124, -0.02158, -0.03745, 1.703, -0.1913,
           0.01091, 0.01725, -0.003834, -0.4139, 0.06982, 1.629]
    # data: [1.0, 0, 0, 0, 0, 0,
    #        0, 1.0, 0, 0, 0, 0,
    #        0, 0, 1.0, 0, 0, 0,
    #        0.0992, 0.2148, -0.03932, 0.371, 0.4616, -0.9249,
    #        -0.08834, -0.1070, -0.2797, 0.4335, 0.9466, -1.516,
    #        -0.003491, 0.05097, 0, -0.09557, -0.2023, 0.3558]

kf_estimator:
  Q:
    data: [1, 0, 0, 0, 0, 0,
           0, 1, 0, 0, 0, 0,
           0, 0, 1, 0, 0, 0,
           0, 0, 0, 1.1, 0, 0,
           0, 0, 0, 0, 1.1, 0,
           0, 0, 0, 0, 0, 1.1]
  R:
    data: [0.01, 0, 0, 0, 0.01, 0, 0, 0, 0.01]

pose_controller:
  vd_max: 0.05
  wd_max: 0.25
  position_gain: 0.1
  orientation_gain: 0.1

ects_controller:
  alpha: 0.5
  beta: 1
  inverse_damping: 0.0001

grasp:
  translational_axis: z
  rotational_axis: -x #y

adaptive_estimator:
  alpha_force: -0.01 # 0.01
  beta_force: -0.0
  force_slack: 0.5
  alpha_adapt_t: 0
  alpha_torque: 0.1
  beta_torque: -0.0
  alpha_adapt_r: 0
  torque_slack: 0.05
  max_force: 5
  max_torque: 0.2
